### PR TITLE
api: port BTCSignMessageRequest to Rust 

### PR DESCRIPTION
**Pull request by benma**
This is very similar to https://github.com/digitalbitbox/bitbox02-firmware/pull/681, just for BTC instead of for ETH.

### CODE REVIEW

File: src/rust/bitbox02-rust/src/shiftcrypto.bitbox02.rs [hunk 1]
```diff
@@ -1,103 +1,99 @@
 #[derive(Clone, PartialEq, ::prost::Message)]
 pub struct PubResponse {
-    #[prost(string, tag="1")]
+    #[prost(string, tag = "1")]
```
[the remark this answers is not in the document]
Comment (x1ddos) [reply src/rust/bitbox02-rust/src/shiftcrypto.bitbox02.rs:3]: When I build locally from the tip of this PR in the docker container, the shiftcrypto.bitbox02.rs is modified and I get the diff back like it was before:

image

[the remark this answers is not in the document]
Comment (x1ddos) [reply src/rust/bitbox02-rust/src/shiftcrypto.bitbox02.rs:3]:
```
$ git rev-parse HEAD
6f98bb9a389858bd9afe7e8f0d9319c6d24a577a
```

**Review by x1ddos — APPROVED** [review]
Very nice. Tried signing a msg with master firmware and this version. Match exactly except electrons had a slightly different spin. :)

Apart from that all good but not sure why shiftcrypto.bitbox02.rs is changed.

File: src/rust/bitbox02-rust/src/shiftcrypto.bitbox02.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (benma) [reply src/rust/bitbox02-rust/src/shiftcrypto.bitbox02.rs:3]: My bad, I had this opened and saved it, so it got auto formatted. Will revert.

**Pull request merged**